lldb/unittests/Process/elf-core/CMakeLists.txt

```diff
@@ -1,3 +1,14 @@
+include(CheckSymbolExists)
+include(CMakePushCheckState)
+
+cmake_push_check_state()
+set(CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
+check_symbol_exists(gettid "unistd.h" HAVE_GETTID)
+if(HAVE_GETTID)
+  add_compile_definitions(-DHAVE_GETTID)
```

Comment:
You can make this more targeted by applying it only to `ProcessElfCoreTests` with `target_compile_definitions`, but either way is fine with me.